api/sys_plugins.go

```diff
@@ -32,10 +32,11 @@ type ListPluginsResponse struct {
 }
 
 type PluginDetails struct {
-	Type    string `json:"string"`
-	Name    string `json:"name"`
-	Version string `json:"version,omitempty"`
-	Builtin bool   `json:"builtin"`
+	Type              string `json:"string"`
+	Name              string `json:"name"`
+	Version           string `json:"version,omitempty"`
+	Builtin           bool   `json:"builtin"`
```

Comment:
Would it make sense to add `mapstructure` tags for all these fields like we did for `DeprecationStatus`?

Reply:
I thought about this, but I think it's maybe  out of scope for this PR, as I'd have to do some testing to see the effects. I made this change for DeprecationStatus, because the status was not being picked up in the client response without it. Maybe I should dig into why deprecation status is unique in this way...

Reply:
What's the command to reproduce the problem you saw? Just `vault plugin list -detailed`?

Reply:
Okay, this appears to be related to the way `mapstructure` handles underscores in `json` tags. To fix, we can either drop the underscore from the tag, or define a new decoder+tag name (`json`). I've gone with the latter fix and dropped the `mapstructure` tag here.

Reply:
On second thought, I think I prefer having the `mapstructure` tag here and no decoder. This will allow all `mapstructure` decoders to handle this without having to specify a custom decoder... This pattern is also used elsewhere in the code.